test_conformance/extensions/cl_khr_external_semaphore/test_external_semaphore.cpp

```diff
@@ -96,18 +96,26 @@ static cl_int get_device_semaphore_handle_types(
 
     num_handle_types =
         size_handle_types / sizeof(cl_external_semaphore_handle_type_khr);
-    std::vector<cl_external_semaphore_handle_type_khr>
-        handle_types_query_result(num_handle_types);
-    err = clGetDeviceInfo(deviceID, param,
-                          handle_types_query_result.size()
-                              * sizeof(cl_external_semaphore_handle_type_khr),
-                          handle_types_query_result.data(), nullptr);
-    test_error(err, "Failed to get exportable handle types");
-
-    for (auto handle_type : handle_types_query_result)
+
+    // Empty list (size_handle_types:0) is a valid value denoting that
+    // implementation is incapable to import/export semaphore
+    if (num_handle_types > 0)
     {
-        handle_types.push_back(handle_type);
+        std::vector<cl_external_semaphore_handle_type_khr>
+            handle_types_query_result(num_handle_types);
+        err =
+            clGetDeviceInfo(deviceID, param,
+                            handle_types_query_result.size()
+                                * sizeof(cl_external_semaphore_handle_type_khr),
+                            handle_types_query_result.data(), nullptr);
```

Comment:
Minor: could we read directly into the `handle_types` vector here, rather than reading into a temporary vector and then copying?